flink-table/flink-table-runtime-blink/src/main/java/org/apache/flink/table/dataformat/BinaryArray.java

```diff
@@ -191,6 +191,23 @@ public Decimal getDecimal(int pos, int precision, int scale) {
 		return Decimal.readDecimalFieldFromSegments(segments, offset, offsetAndSize, precision, scale);
 	}
 
+	@Override
+	public SqlTimestamp getTimestamp(int pos, int precision) {
+		assertIndexIsValid(pos);
+
+		if (SqlTimestamp.isCompact(precision)) {
+			return SqlTimestamp.fromEpochMillis(
+				SegmentsUtil.getLong(segments, getElementOffset(pos, 8)));
+		}
+
+		int fieldOffset = getElementOffset(pos, 8);
+		final long offsetAndNanoOfMilli = SegmentsUtil.getLong(segments, fieldOffset);
+		final int nanoOfMillisecond = (int) offsetAndNanoOfMilli;
```

Comment:
Can you add a static util `readTimestampFieldFromSegments` to `SqlTimestamp`?

Reply:
Sure